explorer-ui/src/components/ProfileView/Preferences/Preferences.jsx

```diff
@@ -14,176 +14,257 @@ import Geocode from "react-geocode";
   const [touristAttractions, setTouristAttractions] = useState([]);
-  const [currLocation, setCurrLocation] = useState({});
+  const [currLocation, setCurrLocation] = useState(null);
   const [budget, setBudget] = useState();
   const [visibility, setVisibility] = useState(null);
-  const [position, setPosition] = useState({"lat": null, "lng":null})
+  const [position, setPosition] = useState({ lat: null, lng: null });
   const budgetOption = React.createRef();
   const visibilityOption = React.createRef();
   const employmentOption = React.createRef();
-  const country = React.createRef();
-  const province = React.createRef();
-  const city = React.createRef();
+  const region = React.createRef();
 
-//TODO: Figure out how to set all the states at once? and add useEffects to all the function calls
-// Update the state array for the users interests
-function handleInterestSelected(event, interest) {
+  //TODO: Figure out how to set all the states at once? and add useEffects to all the function calls
+  // Update the state array for the users interests
+  function handleInterestSelected(event, interest) {
     event.preventDefault();
-    setInterests(interests => [...interests, interest])
-    setPreferenceInfo({"interests": interests, "attractions": preferenceInfo.attractions, "occupation": preferenceInfo.occupation, "budget": preferenceInfo.budget, "currLocation": preferenceInfo.currLocation, "visibility": preferenceInfo.visibility})
-    console.log("preferenceInfo: ", preferenceInfo)
-    console.log("interests so far: ", interests)
-  };
+    setInterests((interests) => [...interests, interest]);
+    setPreferenceInfo({
+      interests: interests,
+      attractions: preferenceInfo.attractions,
+      occupation: preferenceInfo.occupation,
+      budget: preferenceInfo.budget,
+      currLocation: preferenceInfo.currLocation,
+      visibility: preferenceInfo.visibility,
+    });
+    console.log("preferenceInfo: ", preferenceInfo);
+    console.log("interests so far: ", interests);
+  }
 
-// Update the state array for the users attractions interests
-function handleAttractionsSelected(event, attraction) {
+  // Update the state array for the users attractions interests
+  function handleAttractionsSelected(event, attraction) {
     event.preventDefault();
-    setTouristAttractions(touristAttractions => [...touristAttractions, attraction])
-    setPreferenceInfo({"interests": preferenceInfo.interests, "attractions": touristAttractions, "occupation": preferenceInfo.occupation, "budget": preferenceInfo.budget, "currLocation": preferenceInfo.currLocation, "visibility": preferenceInfo.visibility})
-    console.log("preferenceInfo: ", preferenceInfo)
-    console.log("attractions so far: ", touristAttractions)
-  };
+    setTouristAttractions((touristAttractions) => [
+      ...touristAttractions,
+      attraction,
+    ]);
+    setPreferenceInfo({
+      interests: preferenceInfo.interests,
+      attractions: touristAttractions,
```

Comment:
since setting this value before -> would not be the updated touristAttractions state

would flip the order !

Reply:
fixed